external/external.sh

```diff
@@ -436,10 +439,15 @@ if [ $command_type == "load" ]; then
 	fi
 fi
 
-if [ $command_type == "clean" ]; then
+if [ "${command_type}" == "clean" ] ; then
 	if [[ ${test} == 'external_custom' ]]; then
 			test="$(echo ${EXTERNAL_CUSTOM_REPO} | awk -F'/' '{print $NF}' | sed 's/.git//g')"
 	fi
+	if [ "${EXTERNAL_AQA_CLEAN}" == "false" ] ; then
```

Comment:
what is the default of EXTERNAL_AQA_CLEAN? How would it be changed?

Reply:
It is the variable I introduced. If y oou have any better idea I would happily use it.
The makefiles are calling the clean at the end, and I it is very often undesired to lost the results. I have not found a better way how to propagate the "dont clean" to the makefiles (which are calling this script).

One other way led to several if statemnts around calling of the external.sh. The direct variable seemed better. I had also wrote it to the header and readme.

Reply:
It can be changed as any other env. variable.

Reply:
For tkg story env variable has to be visible and set by build.xml. See other env variables in [build.xml](https://github.com/adoptium/aqa-tests/blob/master/external/build.xml)

Reply:
This is  not running in container. This is the part which is launching the contianer (just to ensure we are bothj in same page).


This variable is not set nowhere. It is up to user **of build.xml** to slightly modify the behaviour.  You do not want to run the runs we clean off. You want to run clean on. However for debugging, it was of immense help, to disallow `make _target ` to clean after itself pretty often.

Reply:
Yes, I understand it's the part of launching the container. Take /lucene-solr for example, after the test run the image is cleaned by [$(TEST_ROOT)$(D)external$(D)external.sh --clean --tag "${DOCKERIMAGE_TAG}" --dir lucene-solr](https://github.com/adoptium/aqa-tests/blob/master/external/lucene-solr/playlist.xml#L20). My question is how the variable EXTERNAL_AQA_CLEAN be set and visible to make _lucene_solr_nightly_smoketest? To support `If EXTERNAL_AQA_CLEAN is false, then the image is not cleaned` I think more changes are needed. Maybe you can separate that from this PR.

Reply:
something like this:
```
EXTERNAL_AQA_CLEAN=false make _lucene_solr_nightly_smoketest
or
EXTERNAL_AQA_CLEAN=false  make _jacoco_test
```

works perfectly fine. Aka the final images are not clean after the run. Very good thing is, that they are cleared on next startup anyway.
If this is kept, I can then 
```
podman run --privileged  -v /usr/lib/jvm/java-11-openjdk-11.0.19.0.7-1.fc37.portable.jdk.x86_64/:/opt/java/openjdk  --name jacoco-test --rm -i --entrypoint /bin/bash  adoptopenjdk-jacoco-test:11-jdk-ubuntu-hotspot-full
```
Moreover out of the box, and immediately debug without any more interventions.

Thanx!

Reply:
I see. `EXTERNAL_AQA_CLEAN=false make _lucene_solr_nightly_smoketest` won't work for jenkins story without other changes. It will work locally and seems mainly needed for debug. Suggest update docs.